fmriprep/utils/bids.py

```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# emacs: -*- mode: python; py-indent-offset: 4; indent-tabs-mode: nil -*-
+# vi: set ft=python sts=4 ts=4 sw=4 et:
+"""
+Utilities to handle BIDS inputs
+"""
+import os.path as op
+import warnings
+from glob import glob
+from copy import deepcopy
+from bids.grabbids import BIDSLayout
+
+INPUTS_SPEC = {'fieldmaps': [], 'func': [], 't1': [], 'sbref': [], 't2w': []}
+
+
+def collect_participants(bids_dir, participant_label=None):
+    """
+    Lists the participants under the BIDS root and checks that participants
+    designated with the participant_label argument exist in that folder.
+
+    Returns the list of participants to be finally processed.
+
+
+    """
+    bids_dir = op.abspath(bids_dir)
+    all_participants = [op.basename(subdir)[4:]
+                        for subdir in glob(op.join(bids_dir, 'sub-*'))]
+
+    # Error: bids_dir does not contain subjects
+    if not all_participants:
+        raise RuntimeError(
+            'Could not find participants in "{}". Please make sure the BIDS data '
+            'structure is present and correct. Datasets can be validated online '
+            'using the BIDS Validator (http://incf.github.io/bids-validator/).\n'
+            'If you are using Docker for Mac or Docker for Windows, you '
+            'may need to adjust your "File sharing" preferences.'.format(bids_dir))
+
+    if participant_label is None or not participant_label:
+        return all_participants
+
+    # Drop sub- prefixes
+    participant_label = [sub[4:] if sub.startswith('sub-') else sub for sub in participant_label]
+    # Remove duplicates
+    participant_label = list(set(participant_label))
+    # Remove labels not found
+    found_label = list(set(participant_label) & set(all_participants))
+    if not found_label:
+        raise RuntimeError('Could not find participants [{}] in folder '
+                           '"{}".'.format(', '.join(participant_label), bids_dir))
+
+    notfound_label = list(set(participant_label) - set(all_participants))
+    if notfound_label:
+        warnings.warn('Some participants were not found: {}'.format(
+            ', '.join(notfound_label)), RuntimeWarning)
+    return found_label
```

Comment:
Do we want the outputs (and exception/warning messages) of this function to be sorted? Feel like if I'm looking through logs, it'll be easier to see an issue in a list of participant labels if they're sorted than some random order generated by `set` or `glob`.